lib/brains/tests/test_webhook.py

```diff
@@ -88,37 +107,87 @@ def test_post_auth_fails(self):
         self.req.post.return_value = self.get_response('', 403)
         eq_(self.client.post(self.url, data=example()).status_code, 422)
 
-    def test_post_ok(self):
-        self.req.post.return_value = self.get_response('foo', 204)
-        with patch('lib.brains.views.webhook.XmlUtil.dict_from_xml') as res:
-            res.return_value = {
-                'notification': {'kind': '', 'subject': subscription()}}
-            with patch('lib.brains.views.webhook.Processor.process'):
-                eq_(self.client.post(self.url, data=example()).status_code,
-                    204)
-
 
-class SubscriptionTest(BraintreeTest):
+@override_settings(BRAINTREE_PROXY='http://m.o')
+class TestWebhookWithSubscription(SubscriptionTest):
 
     def setUp(self):
-        self.buyer, self.braintree_buyer = create_braintree_buyer()
-        self.method = create_method(self.braintree_buyer)
-        self.seller, self.seller_product = create_seller()
-        self.braintree_sub = BraintreeSubscription.objects.create(
-            paymethod=self.method,
-            seller_product=self.seller_product,
-            provider_id='some-bt:id'
-        )
-        self.sub = subscription(id='some-bt:id')
+        super(TestWebhookWithSubscription, self).setUp()
+        self.url = reverse('braintree:webhook')
+        self.patch_webhook_forms()
+        self.req.post.return_value = self.get_response('foo', 204)
+
+    def test_post_ok(self):
+        with patch('lib.brains.views.webhook.XmlUtil.dict_from_xml') as attr:
+            attr.return_value = {
+                'notification': {
+                    'kind': 'subscription_charged_successfully',
+                    'subject': subscription()
+                }
+            }
+            res = self.client.post(self.url, data=example())
+            eq_(res.status_code, 200)
+            eq_(res.json.keys(), ['mozilla', 'braintree'])
+
+    def test_post_ignored(self):
+        with patch('lib.brains.views.webhook.XmlUtil.dict_from_xml') as attr:
+            attr.return_value = {
+                'notification': {
+                    'kind': '',
+                    'subject': ''
+                }
+            }
+            res = self.client.post(self.url, data=example())
+            eq_(res.status_code, 204)
 
 
 class TestSubscription(SubscriptionTest):
+    kind = 'subscription_charged_successfully'
 
     def process(self, subscription):
         process = Processor(notification(subject=subscription))
         process.update_transactions(process.webhook.subscription,
                                     self.braintree_sub)
 
+    def test_data(self):
+        process = Processor(notification(
+            subject=subscription(), kind=self.kind))
+        process.process()
```

Comment:
since this is two lines of code now, might be a good excuse to replace it with ``self.process()``